### PR TITLE
Add mac CI

### PR DESCRIPTION
**Pull request by RikkiGibson**

### CODE REVIEW

**Comment by RikkiGibson** [comment]
Draft PRs aren't kicking CI at the moment using the "For Personal Review Only" tag instead.

File: src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs [hunk 1]
```diff
@@ -157,6 +158,7 @@ public static class ExecutionConditionUtil
 
         public static bool IsWindows => Path.DirectorySeparatorChar == '\\';
         public static bool IsUnix => !IsWindows;
+        public static bool IsMacOS => RuntimeInformation.IsOSPlatform(OSPlatform.OSX); // NOTE: may not be valid on mono
```
Comment (RikkiGibson) [inline src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]: Not sure if we want to keep this comment or not. If there's an actual problem people need to be aware of let's link to a tracking issue.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -215,3 +215,35 @@ jobs:
       continueOnError: true
       condition: not(succeeded())
 
+- job: macOS_Test
+  pool:
+    vmImage: 'macOS-10.13'
+  strategy:
+    maxParallel: 2
```
Comment (jaredpar) [inline azure-pipelines.yml]: don't need this as there is one job

File: azure-pipelines.yml [hunk 3]
```diff
@@ -215,3 +215,35 @@ jobs:
...
+    matrix:
+      coreclr:
+        _args: --testCoreClr
+        _name: CoreClr
+        _configuration: Debug
```
Comment (jaredpar) [inline azure-pipelines.yml]: i'd delete the configuration variables here and just inline as there is only one job here.

File: src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs [hunk 4]
```diff
@@ -100,10 +100,12 @@ void tryCompileDeepFluentCalls(int depth)
 
         [Fact]
         [WorkItem(33909, "https://github.com/dotnet/roslyn/issues/33909")]
+        [WorkItem(34880, "https://github.com/dotnet/roslyn/issues/34880")]
         public void DeeplyNestedGeneric()
         {
             int nestingLevel = (ExecutionConditionUtil.Architecture, ExecutionConditionUtil.Configuration) switch
             {
+                _ when ExecutionConditionUtil.IsMacOS => 100,
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Emit/Emit/EndToEndTests.cs]: This number seems pretty low even for debug. But I did find that 170 was too high.

File: src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs (hunk 1 above)
Comment (jaredpar) [reply src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]: Lets delete the comment. There is no known active bug here.

File: src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs [hunk 5]
```diff
@@ -253,6 +255,12 @@ public class UnixLikeOnly : ExecutionCondition
         public override string SkipReason => "Test not supported on Windows";
     }
 
+    public class WindowsOrLinuxOnly : ExecutionCondition
+    {
+        public override bool ShouldSkip => ExecutionConditionUtil.IsMacOS;
+        public override string SkipReason => "Test not supported on macOS";
```
Comment (jaredpar) [inline src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]:
```suggestion
        public override string SkipReason => "Test not supported on MacOS";
```

File: src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs (hunk 1 above)
Comment (RikkiGibson) [reply src/Test/Utilities/Portable/Assert/ConditionalFactAttribute.cs]: Done

File: azure-pipelines.yml (hunk 2 above)
Comment (RikkiGibson) [reply azure-pipelines.yml]: Done

File: azure-pipelines.yml (hunk 3 above)
Comment (RikkiGibson) [reply azure-pipelines.yml]: Done

**Comment by RikkiGibson** [comment]
Odd, the mac run seems to have instantly blown up after pushing that config change. Maybe a transient issue?

**Pull request merged**